hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/BucketManagerImpl.java

```diff
@@ -466,26 +486,29 @@ public boolean removeAcl(OzoneObj obj, OzoneAcl acl) throws IOException {
         throw new OMException("Bucket " + bucket + " is not found",
             BUCKET_NOT_FOUND);
       }
-      List<OzoneAcl> list = bucketInfo.getAcls();
-      if (!list.contains(acl)) {
-        // Return false if acl doesn't exist in current ACLs.
-        LOG.info("Acl:{} not found in existing ACLs:{}.", acl,
-            StringUtils.join(",", list));
-        return false;
+
+      // When we are removing subset of rights from existing acl.
+      for(OzoneAcl a: bucketInfo.getAcls()) {
+        if(a.getName().equals(acl.getName()) &&
```

Comment:
should we return false if no such acl is found?

Reply:
Also notice the difference in the behavior in OmOzoneAclMap.java where add existing and remove non-existing throw exception. Can you elaborate the difference?

Reply:
Behavior in VolumeManagerImpl is same. We catch the exception thrown from OmOzoneAclMap and then return false. Exception is thrown from OmOzoneAclMap as OmVolumeArgs api's return type is void.

Reply:
Had offline discussion with @xiaoyuyao , we will change log statement in VolumeManagerImpl to debug.